File: Proyecto/Calendario_App/src/main/java/org/example/calendario_app/model/Cliente.java
package org.example.calendario_app.model;

import lombok.Data;

import java.time.LocalDate;

@Data
public class Cliente {
    private int id_cliente;
    private String nombre;
    private String apellidos;
    private LocalDate fecha_creacion;
    private LocalDate fecha_modificacion;

    public Cliente(String nombre, String apellidos, LocalDate fecha_creacion) {
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.fecha_creacion = fecha_creacion;
    }

    public Cliente(int id_cliente, String nombre, String apellidos, LocalDate fecha_creacion,
            LocalDate fecha_modificacion) {
        this.id_cliente = id_cliente;
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.fecha_creacion = fecha_creacion;
        this.fecha_modificacion = fecha_modificacion;
    }

    public int getId() {
        return id_cliente;
    }

    @Override
    public String toString() {
        return "Cliente{" +
                "id_cliente=" + id_cliente +
                ", nombre='" + nombre + '\'' +
                ", apellidos='" + apellidos + '\'' +
                ", fecha_creacion=" + fecha_creacion +
                ", fecha_modificacion=" + fecha_modificacion +
                '}';
    }
}

File: Proyecto_Android/Calendario_Android_App-31-01-2026/Calendario_Android_App-31-01-2026/Calendario_Android_App/app/src/main/java/com/example/calendario_android_app/model/Cliente.kt
package com.example.calendario_android_app.model

import java.sql.Timestamp

/**
 * Modelo de datos que representa un cliente en el sistema.
 * Los clientes son personas físicas que pueden tener una cuenta de usuario asociada.
 * 
 * @property id_cliente Identificador único del cliente
 * @property nombre Nombre del cliente
 * @property apellidos Apellidos del cliente
 * @property creado_el Fecha y hora de creación del registro
 * @property actualizado_el Fecha y hora de la última actualización
 */
data class Cliente(
    val id_cliente: Int,
    val nombre: String,
    val apellidos: String,
    val creado_el: Timestamp,
    val actualizado_el: Timestamp
)
